funding/manager.go

```diff
@@ -842,21 +854,47 @@ func (f *Manager) CancelPeerReservations(nodePub [33]byte) {
 	delete(f.activeReservations, nodePub)
 }
 
+// chanIdentifier wraps pending channel ID and channel ID into one struct so
+// it's easier to identify a specific channel.
+type chanIdentifier struct {
+	// tempChanID is the pending channel ID created by the funder when
+	// initializing the funding flow. For fundee, it's received from the
+	// `open_channel` message.
+	tempChanID lnwire.ChannelID
+
+	// chanID is the channel ID created by the funder once the
+	// `accept_channel` message is received. For fundee, it's received from
+	// the `funding_created` message.
+	chanID lnwire.ChannelID
+}
+
 // failFundingFlow will fail the active funding flow with the target peer,
 // identified by its unique temporary channel ID. This method will send an
 // error to the remote peer, and also remove the reservation from our set of
 // pending reservations.
 //
 // TODO(roasbeef): if peer disconnects, and haven't yet broadcast funding
 // transaction, then all reservations should be cleared.
-func (f *Manager) failFundingFlow(peer lnpeer.Peer, tempChanID [32]byte,
+func (f *Manager) failFundingFlow(peer lnpeer.Peer, cid *chanIdentifier,
 	fundingErr error) {
 
 	log.Debugf("Failing funding flow for pending_id=%x: %v",
-		tempChanID, fundingErr)
+		cid.tempChanID, fundingErr)
+
+	// First, notify Brontide to remove the pending channel.
+	//
+	// NOTE: depending on where we fail the flow, we may not have the
+	// active channel ID yet.
+	if !cid.chanID.IsEmpty() {
```

Comment:
Maybe just have a bool in `chanIdentifier` instead? What if somebody decides to use an all-zero ChannelID for a zero-conf channel?

Reply:
We send to brontide for zero-conf channels. I think it might just be better to have a bool that each caller sets if we should call `RemovePendingChannel` since that seems a lot harder to mess up / no hidden gotchas

Reply:
cool yeah i like the idea. Change to be more explicit!